### PR TITLE
perf(aad-manifest): allow user to config frontend and bot info

### PR DESCRIPTION
**Pull request by SLdragon**
1. Allow user to config frontend and bot info for existing app, user config has the highest priority:

```
{
    "$schema": "https://aka.ms/teamsfx-env-config-schema",
    "description": "You can customize the TeamsFx config for different environments. Visit https://aka.ms/teamsfx-env-config to learn more about this.",
    "manifest": {
        "appName": {
            "short": "test-local-debug",
            "full": "Full name for test-local-debug"
        }
    },
    "auth": {
        "frontendDomain": "example.com",
        "botId": "xxxx",
        "botEndpoint": "https://example.com"
    }
}
```

2. Skip arm deployment if there is no Azure resource need to provision
3. Update config json schema

### CODE REVIEW

**Comment by SLdragon** [comment]
Will add and update test case in following commit

File: packages/fx-core/src/plugins/solution/fx-solution/v2/utils.ts [hunk 1]
```diff
@@ -58,6 +59,23 @@ export function isAzureProject(azureSettings: AzureSolutionSettings | undefined)
   return azureSettings !== undefined && HostTypeOptionAzure.id === azureSettings.hostType;
 }
 
+export function hasAzureResourcesToProvision(
```
Comment (jayzhang) [inline packages/fx-core/src/plugins/solution/fx-solution/v2/utils.ts:62]: Can we reuse this method?
image

Comment (SLdragon) [reply packages/fx-core/src/plugins/solution/fx-solution/v2/utils.ts:62]: Thanks, will use this~

File: packages/fx-core/src/plugins/resource/aad/plugin.ts [hunk 2]
```diff
@@ -232,17 +233,37 @@ export class AadAppForTeamsImpl {
...
+      ) {
+        throw ResultFactory.UserError(AppIdUriInvalidError.name, AppIdUriInvalidError.message());
+      }
     }
 
-    let applicationIdUri = "api://";
-    applicationIdUri += config.frontendDomain ? `${config.frontendDomain}/` : "";
-    applicationIdUri += config.botId ? "botid-" + config.botId : config.clientId;
-    config.applicationIdUri = applicationIdUri;
+    config.frontendDomain = userSetFrontendDomain ?? config.frontendDomain;
```
Comment (KennethBWSong) [inline packages/fx-core/src/plugins/resource/aad/plugin.ts:250]: should we validate user input?

Comment (SLdragon) [reply packages/fx-core/src/plugins/resource/aad/plugin.ts:257]: Thanks, updated

**Review by KennethBWSong — DISMISSED** [review]

**Review by jayzhang — APPROVED** [review]

File: packages/api/src/schemas/envConfig.json [hunk 3]
```diff
@@ -32,6 +32,21 @@
           "type": "string",
           "description": "The access_as_user scope id of existing AAD app for Teams app.",
           "minLength": 1
+        },
+        "frontendDomain": {
+          "type": "string",
+          "description": "The frontend domain for redirect Url of existing AAD app for Teams app.",
```
Comment (qinezh) [inline packages/api/src/schemas/envConfig.json:38]: typo Url => url?

File: packages/api/src/schemas/envConfig.json [hunk 4]
```diff
@@ -32,6 +32,21 @@
...
+          "minLength": 1
+        },
+        "botId": {
+          "type": "string",
+          "description": "The bot id for identifier Uris of existing AAD app for Teams app.",
+          "minLength": 1
+        },
+        "botEndpoint": {
```
Comment (qinezh) [inline packages/api/src/schemas/envConfig.json:46]: add `botId` as dependency since they should be configured in pairs?

Comment (SLdragon) [reply packages/api/src/schemas/envConfig.json:46]: Oh, thanks, will update

File: packages/api/src/schemas/envConfig.json (hunk 3 above)
Comment (SLdragon) [reply packages/api/src/schemas/envConfig.json:38]: Thanks, how about using URL?

Comment (qinezh) [reply packages/api/src/schemas/envConfig.json:38]: Good to me 😄

**Review by qinezh — APPROVED** [review]
LGTM for the env config part.

**Review by KennethBWSong — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Pull request merged**